template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/knox/KnoxServiceConfigProviderTest.java

```diff
@@ -38,9 +38,27 @@ public void testGetServiceTypeShouldReturnKnox() {
         Assert.assertTrue(underTest.getServiceType().equals(KNOX));
     }
 
-    @Test
-    public void testGetRoleConfigsShouldReturnEmptyList() {
-        Assert.assertTrue(underTest.getRoleConfigs("", null).isEmpty());
+    @ParameterizedTest(name = "{index}: check knox properties cm version {0} and cdh version {1} will produce {2} property")
+    @MethodSource("cmCdhCombinationsRoleConfigs")
+    public void testGetRoleConfigsShouldReturnEmptyList(String cdhVersion, String cmVersion, boolean ssl, int numberOfProperties) {
```

Comment:
Besides checking for `numberOfProperties`, it would be useful to explicitly verify the existence / absence of the JDBC URL property depending on `ssl`.